auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -481,26 +493,24 @@ public TypeSpec createTypeAdapter(Context context, ClassName className, ClassNam
         continue;
       }
```

Comment:
Here, we substitute the property type for its type parameter. All downstream processing thereby stays the same for handling parameterized types, primitives, etc.